types/prosemirror-view/index.d.ts

```diff
@@ -48,87 +109,32 @@ export class Decoration {
    * also directly pass a DOM node. getPos can be used to find the
    * widget's current document position.
    */
-  static widget(
+  static widget<T extends {} = { [key: string]: any }>(
```

Comment:
Use `static widget<T extends object = { [key: string]: any }>(` instead. The `extends object` makes sure that one gets a type error if for example one tried to use a string as the `spec` argument:

```typescript
Decoration.widget(0, (x) => x.dom, 'this argument should be an object, not a string!'); // $ExpectError
```

(Same for other functions.)

Reply:
Thanks @timjb! I never knew this. I've just updated several other projects with this enhancement.